cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -345,9 +370,10 @@ func (a *StaticAutoscaler) RunOnce(currentTime time.Time) errors.AutoscalerError
 				a.lastScaleDownDeleteTime = currentTime
 				a.clusterStateRegistry.Recalculate()
 			}
-			if a.processors != nil && a.processors.ScaleDownStatusProcessor != nil {
-				a.processors.ScaleDownStatusProcessor.Process(autoscalingContext, scaleDownStatus)
-			}
+
+			a.processors.ScaleDownStatusProcessor.Process(autoscalingContext, scaleDownStatus)
```

Comment:
Is there any reason we need to process result from here, instead of relying on deferred call?

Reply:
Consistency with the scale-up processor, i.e. if we added some logic between the scale down and the deferred call, this logic should be able to depend on the fact that if a scale down happens, its result is processed immediately after (as is the case with a scale up). Unless we don't need that guarantee for the scale-up processor, then they can both be executed in the deferred call.